dojo/tools/wazuh/wazuh-vulns-extractor.py

```diff
@@ -0,0 +1,120 @@
+import requests
+from requests.auth import HTTPBasicAuth
+import json
+import urllib3
+
+# Suppress InsecureRequestWarning
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+# Wazuh
+BASE_URL = "https://<wazuh-ip-or-url>:55000"
+AUTH_URL = f"{BASE_URL}/security/user/authenticate?raw=true"
+HEADERS = {}
+
+# Basic authentication creds
+USERNAME = "<wazuh-api-user>"
+PASSWORD = "<wazuh-api-pass>"
+
+"""
+Using Two Groups for Wazuh Agent Queries:
+
+- Provides precise targeting by intersecting two specific groups, even if agents belong to multiple groups.
+- Balances between broad and narrow agent selections.
+- If narrow targeting isn't desired, set both group variables to the same name for a wider selection.
+- Additionally, it appends an 'agent_ip' field for every vulnerability. This is later processed by DefectDojo to create endpoints and correlate each vulnerability to a specific agent, enhancing traceability and accountability.
+
+Note: This approach refines the vulnerability reporting process by correlating agents with vulnerabilities more efficiently.
+"""
+
+
+GROUP_1 = "<group1-name>"
+GROUP_2 = "<group2-name>"
+
+# Authenticate and set token
+
+
+def authenticate():
+    response = requests.get(
+        AUTH_URL, auth=HTTPBasicAuth(USERNAME, PASSWORD), verify=False
```

Comment:
Default `verify=False` is not the best. But feel free to enable to set it up via `extras` in `Tool_Configuration`